cmd/intake-receiver/go.mod

```diff
@@ -0,0 +1,17 @@
+module intake-receiver
```

Comment:
I added a `go.mod` file here to avoid updating the `go.elastic.co/apm/v2` dependency in this PR. Ideally we'd update the dependency in a separate PR and merge and remove this module afterwards.